public/react-services/pattern-handler.js

```diff
@@ -22,80 +22,92 @@ export class PatternHandler {
+          });
         }
-      }
 
-      const wazuhConfig = new WazuhConfig();
-      const { pattern } = wazuhConfig.getConfig();
-      const indexPatternFound = patternList.find((indexPattern) => indexPattern.title === pattern);
+        const indexPatternFound = patternList.find(
+          (indexPattern) => indexPattern.title === pattern
+        );
 
-      if (!indexPatternFound) {
-        // if no valid index patterns are found we try to create the wazuh-alerts-*
-        try {
+        if (!indexPatternFound) {
+          // if no valid index patterns are found we try to create the wazuh-alerts-*
+          try {
+            if (!pattern) return;
 
-          if (!pattern) return;
+            getToasts().add({
+              color: 'warning',
+              title: `No ${pattern} index pattern was found, proceeding to create it.`,
+              toastLifeTimeMs: 5000,
+            });
 
-          getToasts().add({
-            color: 'warning',
-            title:
-              `No ${pattern} index pattern was found, proceeding to create it.`,
-            toastLifeTimeMs: 5000
-          });
+            if (await SavedObject.getExistingIndexPattern(pattern)) {
+              await SavedObject.refreshIndexPattern(pattern);
+              getToasts().addSuccess(`${pattern} index pattern updated successfully`);
+            } else {
+              await SavedObject.createWazuhIndexPattern(pattern);
+              getToasts().addSuccess(`${pattern} index pattern created successfully`);
+            }
 
-          await SavedObject.createWazuhIndexPattern(pattern);
-          getToasts().addSuccess(`${pattern} index pattern created successfully`)
-          getDataPlugin().indexPatterns.setDefault(pattern, true);
-        } catch (err) {
-          getToasts().add({
-            color: 'error',
-            title: 'Error creating the index pattern.',
-            text: err.message || err,
-            toastLifeTimeMs: 3000
-          });
-          AppState.removeCurrentPattern();
+            getDataPlugin().indexPatterns.setDefault(pattern, true);
+          } catch (err) {
+            getToasts().add({
+              color: 'error',
+              title: 'Error creating the index pattern.',
+              text: err.message || err,
+              toastLifeTimeMs: 3000,
+            });
+            AppState.removeCurrentPattern();
 
-          this.wzMisc = new WzMisc();
-          this.wzMisc.setBlankScr(
-            'Sorry but no valid index patterns were found and creation was unsuccessful'
-          );
+            this.wzMisc = new WzMisc();
+            this.wzMisc.setBlankScr(
+              'Sorry but no valid index patterns were found and creation was unsuccessful'
+            );
+            if (
+              !window.location.hash.includes('#/settings') &&
+              !window.location.hash.includes('#/blank-screen')
+            ) {
+              window.location.href = '/app/wazuh#/blank-screen/';
```

Comment:
It fails when serving Kibana with `server.basePath` setting set and a proxy. It should use the Kibana service:
```
getHttp().basePath.prepend(url)
```